enclave-boundary/src/untrusted.rs

```diff
@@ -46,7 +46,20 @@ pub fn make_variable_length_ecall(
                 &mut outbuf_used,
                 &mut outbuf_retry_id,
             )
-        } {
+        };
+        if is_fatal_sgx_status(retval) {
+            panic!(
+                "Enclave reported fatal error: ecall retval: {:?}. Panicking to restart.",
+                retval
+            );
+        }
+        if is_fatal_sgx_status(result) {
+            panic!(
+                "Enclave reported fatal error: ecall returned {:?}. Panicking to restart.",
+                result
+            );
+        }
+        match result {
             sgx_status_t::SGX_SUCCESS => match retval {
                 sgx_status_t::SGX_ERROR_OUT_OF_MEMORY => continue,
```

Comment:
If the "pre-check" is done, it needs to look at `result` first. If `result` fails we don't know what `retval` will have in it, it's up to the SDK if it overwrites the SGX_SUCCESS we put in, or if it's going to set it to some other value prior to making the actual enclave call. 

Github comment limitation :(
Suggest keeping the match statement and leveraging [Match Guards](https://doc.rust-lang.org/reference/expressions/match-expr.html#match-guards)
```rust
        match result {
            sgx_status_t::SGX_SUCCESS => match retval {
                sgx_status_t::SGX_ERROR_OUT_OF_MEMORY => continue,
                sgx_status_t::SGX_SUCCESS => {
                    outbuf.truncate(outbuf_used);
                    break Ok(outbuf);
                }
                other_retval if is_fatal_sgx_status(other_retval) => panic!("Enclave reported fatal error: ecall retval: {other_retval:?}. Panicking to restart."),
                other_retval => break (Err(other_retval)),
            },
            status if is_fatal_sgx_status(status) => panic!("Enclave reported fatal error: ecall returned {status:?}. Panicking to restart."),
            status => break Err(status),
        }
```

Reply:
Ooh yeah, match guards! That's more elegant than what I suggested :)

Reply:
Oh, neat, I didn't know about this feature